packages/react-charting/src/components/HorizontalBarChart/HorizontalBarChart.base.tsx

```diff
@@ -288,19 +295,47 @@ export class HorizontalBarChartBase extends React.Component<IHorizontalBarChartP
     );
   }
 
+  // Adjusts the margin percentage, based on the width of the svg (inversly propotional)
+  private _adjustBarSpacing(): void {
+    const svgWidth = this.barChartSvgRef.current?.getBoundingClientRect().width;
+    if (svgWidth) {
+      // [0,1] Controls the spacing between the bars
+      const BAR_SPACING_FACTOR = 1;
+      // The barspacing is provided in percentage terms, so changes inversely to the width of the svg
+      const currentBarSpacing = clamp((200 / svgWidth) * BAR_SPACING_FACTOR, 1, 0.2);
+      this.setState({ barSpacing: currentBarSpacing });
+    }
+  }
+
+  /**
+   * This functions returns an array of <rect> elements, which form the bars
+   * For each bar an x value, and a width needs to be specified
+   * The computations are done based on percentages
+   * Extra margin is also provided, in the x value to provide some spacing in between the bars
+   */
+
   private _createBars(data: IChartProps, palette: IPalette): JSX.Element[] {
+    const noOfBars =
+      data.chartData?.reduce(
+        (count: number, point: IChartDataPoint) => (count += (point.horizontalBarChartdata?.x || 0) > 0 ? 1 : 0),
+        0,
+      ) || 1;
+    const totalSpacing = this.state.barSpacing * (noOfBars - 1);
     const defaultPalette: string[] = [palette.blueLight, palette.blue, palette.blueMid, palette.red, palette.black];
     // calculating starting point of each bar and it's range
     const startingPoint: number[] = [];
-    const total = data.chartData!.reduce(
+    let total = data.chartData!.reduce(
       (acc: number, point: IChartDataPoint) =>
         acc + (point.horizontalBarChartdata!.x ? point.horizontalBarChartdata!.x : 0),
       0,
     );
+
+    total += (totalSpacing * total) / (100 - totalSpacing);
+
     let prevPosition = 0;
     let value = 0;
 
-    let sumOfPercent = 0;
+    let sumOfPercent = totalSpacing;
```

Comment:
Are all the ratios adding up to 100 at the end.

Reply:
Yes I believe the ratios will add to 100% (or 99.9999%), since the scale down to widths was computed as per the above calculations already.

I have cross checked in the local development setup by removing the AbsoluteScaleVariant. This is how it looked.